src/Microsoft.Health.Fhir.Subscriptions/Channels/RestHookChannel.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Health.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Features.Routing;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.Fhir.Subscriptions.Models;
+using Microsoft.Health.Fhir.Subscriptions.Validation;
+
+namespace Microsoft.Health.Fhir.Subscriptions.Channels
+{
+    [ChannelType(SubscriptionChannelType.RestHook)]
+
+    #pragma warning disable CA1001 // Types that own disposable fields should be disposable
+    public class RestHookChannel : ISubscriptionChannel
+    #pragma warning restore CA1001 // Types that own disposable fields should be disposable
+    {
+        private readonly ILogger<RestHookChannel> _logger;
+        private readonly IBundleFactory _bundleFactory;
+        private readonly IRawResourceFactory _rawResourceFactory;
+        private readonly HttpClient _httpClient;
+        private readonly IModelInfoProvider _modelInfoProvider;
+        private readonly IUrlResolver _urlResolver;
+        private RequestContextAccessor<IFhirRequestContext> _contextAccessor;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IActionContextAccessor _actionContextAccessor;
+
+        public RestHookChannel(ILogger<RestHookChannel> logger, HttpClient httpClient, IBundleFactory bundleFactory, IRawResourceFactory rawResourceFactory, IModelInfoProvider modelInfoProvider, IUrlResolver urlResolver, RequestContextAccessor<IFhirRequestContext> contextAccessor, IHttpContextAccessor httpContextAccessor, IActionContextAccessor actionContextAccessor)
+        {
+            _logger = logger;
+#pragma warning disable CA2000 // Dispose objects before losing scope
+            _httpClient = new HttpClient(new HttpClientHandler() { CheckCertificateRevocationList = true }, disposeHandler: true);
+#pragma warning restore CA2000 // Dispose objects before losing scope
+            _bundleFactory = bundleFactory;
+            _rawResourceFactory = rawResourceFactory;
+            _modelInfoProvider = modelInfoProvider;
+            _urlResolver = urlResolver;
+
+            var fhirRequestContext = new FhirRequestContext(
+                method: OperationsConstants.Reindex,
+                uriString: "$reindex",
```

Comment:
We should update this context object to Subscription instead of Reindex.  Or perhaps we can remove it, I'm not sure we need it in the end.

Reply:
I think since we might not need this in the end, I replaced them will empty/null values.